cockroachcloud/cmek.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Customer-Managed Encryption Keys (CMEK)
+summary: Use cryptographic keys that you manage to protect data at rest in a CockroachDB Dedicated cluster.
+toc: true
+docs_area: manage.security
+---
+
+Customer-Managed Encryption Keys (CMEK) give you more control over how a {{ site.data.products.dedicated }} cluster's data is protected at rest on the cluster's nodes. When CMEK  enabled, your cluster's data is protected by an additional cryptographic key that is entirely within your control, hosted in a supported key-management systems (KMS) platform. This key is called the _CMEK key_. You grant your cluster's service account the permission to encrypt and decrypt using the CMEK key. The CMEK key is never persisted in a cluster, and {{ site.data.products.db }} uses the KMS platform's API to encrypt and decrypt using the CMEK key. The service account never has direct access to the CMEK key materials and does not have the ability to manage or export your CMEK keys.
+
+You can create and manage your CMEK keys in AWS KMS, Google Cloud KMS, and Hashicorp Vault Secrets Manager. Hashicorp Vault Secrets Manager enables you to access CMEK keys stored in multiple KMS platforms. You can learn more about {{ site.data.products.dedicated }}'s [Hashicorp Vault Integration](/docs/{{site.versions["stable"]}}/hashicorp-integration.html).
+
+This article describes how CMEK works in {{ site.data.products.dedicated }} clusters. To configure CMEK, see [Managing Customer-Managed Encryption Keys (CMEK) for CockroachDB Dedicated](managing-cmek.html).
+
+## Overview of CMEK
+
+This section describes how CMEK works on a {{ site.data.products.dedicated }} cluster. To learn more about encryption in CockroachDB, see [Encryption At Rest](/docs/{{site.versions["stable"]}}/encryption.html). To enable CMEK, see [Managing Customer-Managed Encryption Keys (CMEK) for CockroachDB Dedicated](managing-cmek.html).
+
+When you create a {{ site.data.products.dedicated }} cluster, two encryption keys are created automatically:
+
+1. The _store key_ is a Key Encryption Key (KEK), and is used to protect the data key. The store key is accessed each time a cluster node is started.
+1. The _data key_ is a Data Encryption Key (DEK), and is used to encrypt and decrypt cluster data before it is read from or written to a cluster node's filesystem. Each time the cluster is started or restarted, and each time a node joins the cluster, CockroachDB uses the store key to encrypt and decrypt data keys.
+
+  By default, data keys are rotated weekly, and CockroachDB keeps a registry of all data keys that have ever been used to write data to the cluster. When data is read from the cluster, it is first decrypted using the data key that was used to encrypt it. When data is written to a cluster, it is first encrypted using the current data key.
```

Comment:
We don't need the CMEK key because we call encrypt and decrypt using the KMS's API. It's the KMS's API that does the permission checks to make sure we have access. At cluster start-up we load the encrypted store key into memory and call the KMS's API to decrypt it. Then we can encrypt and decrypt data keys just like we did before. When we rotate the store key, we use the CMEK key to encrypt and decrypt it too, and it begins encrypting and decrypting new data keys. Does that make sense? If so, I'll try to make this section more clear.

Reply:
I've rewritten and de-duplicated this.